### ISSUE TITLE
customAttributeMetadata GraphQl query has no cache identity

### ISSUE THREAD

**Issue by marciomaciel** [magento/magento2#32909]
*As a mobile device I want to have customAttributeMetadata query cached by full page cache and Varnish so we can get a cached version of result without overload our infrastructure during a large volume of requests to obtain details and options of EAV catalog attributes.*

### AC
* Do some requests to customAttributeMetadata query to get attribute options and details for color attribute
* Cache always missed by full page cache and Varnish

### Approved Schema
```graphql
type Query {
    customAttributeMetadata(attributes: [AttributeInput!]!): CustomAttributeMetadata @resolver(class: "Magento\\EavGraphQl\\Model\\Resolver\\CustomAttributeMetadata") @cache(cacheIdentity: "Magento\\EavGraphQl\\Model\\Resolver\\CustomAttributeMetadata\\Identity")
}
```
### Additional details
In addition, using together with a products query to render a product details page, all request will be missed by full page and varnish cache.

**Comment by engcom-Delta** [comment]
Hi @marciomaciel   ,

Thank you for raising an issue , we are happy to work on this issue . Kindly provide steps to reproduce , issue description, actual results with screenshots.

Regards,

**Comment by damienwebdev** [comment]
I'm going to take a stab at this tomorrow. https://github.com/graycoreio/magento2-custom-attribute-metadata-cache

@cpartica please feel free to merge in whatever I end up writing for the cache identity so long as you maintain the copyright header.

**Comment by danmooney2** [comment]
This has been implemented in [B2B-2259](https://github.com/magento/magento2/search?q=B2B-2259&type=commits) and merged to 2.4-develop.  It will be available for use in 2.4.7.
